Add copy colors button

hokageCV opened the pull request:
## Implement 'Copy CSS' button.

Replaced generate button with copy button. 
Now on copying the css, both themes gets copied to user's clipboard.

## Demonstration video 

https://github.com/user-attachments/assets/6e0c67b5-6fa6-4a5b-8ef2-43a810c80ae5

thisisfranciswu commented:
Hi @hokageCV, thanks for the PR! It looks like your change removes the ability to generate a color palette for a specific color the user enters. That's something I'd like to still be able to do. Can you confirm? Thanks!

hokageCV commented:
Hello, yes it looks like I missed that. Will update the PR.

Shall I add another button below `generate`  for `copy`? Or do you have any other suggestions?

thisisfranciswu commented:
Stacking the buttons is what I would have done, although I'd lead with the
generate button instead.

Thanks!

--
*Francis Wu*
www.thisisfranciswu.com


On Tue, Mar 18, 2025 at 10:49 AM Chaitanya Varu ***@***.***>
wrote:

> Hello, yes it looks like I missed that. Will update the PR.
>
> Shall I add another button below generate for copy? Or do you have any
> other suggestions?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/thisisfranciswu/enterprise-ui-palette-generator/pull/2#issuecomment-2733532524>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAADGJIUY4HEME45OVFGM4T2VAXADAVCNFSM6AAAAABZDQZINCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZTGUZTENJSGQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> .com>
> [image: hokageCV]*hokageCV* left a comment
> (thisisfranciswu/enterprise-ui-palette-generator#2)
> <https://github.com/thisisfranciswu/enterprise-ui-palette-generator/pull/2#issuecomment-2733532524>
>
> Hello, yes it looks like I missed that. Will update the PR.
>
> Shall I add another button below generate for copy? Or do you have any
> other suggestions?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/thisisfranciswu/enterprise-ui-palette-generator/pull/2#issuecomment-2733532524>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAADGJIUY4HEME45OVFGM4T2VAXADAVCNFSM6AAAAABZDQZINCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZTGUZTENJSGQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> .com>
>

hokageCV commented:
Agreed. Generate button and below it could be copy button.

thisisfranciswu commented:
Awesome, thanks!

--
*Francis Wu*
www.thisisfranciswu.com


On Tue, Mar 18, 2025 at 11:38 AM Chaitanya Varu ***@***.***>
wrote:

> Agreed. Generate button and below it could be copy button.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/thisisfranciswu/enterprise-ui-palette-generator/pull/2#issuecomment-2733717980>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAADGJJCZNOYWP7NDG7M7FT2VA4YZAVCNFSM6AAAAABZDQZINCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZTG4YTOOJYGA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> .com>
> [image: hokageCV]*hokageCV* left a comment
> (thisisfranciswu/enterprise-ui-palette-generator#2)
> <https://github.com/thisisfranciswu/enterprise-ui-palette-generator/pull/2#issuecomment-2733717980>
>
> Agreed. Generate button and below it could be copy button.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/thisisfranciswu/enterprise-ui-palette-generator/pull/2#issuecomment-2733717980>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAADGJJCZNOYWP7NDG7M7FT2VA4YZAVCNFSM6AAAAABZDQZINCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZTG4YTOOJYGA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> .com>
>

hokageCV committed:
Add copy colors button

Remove colors from the css rules before generating new one to prevent
duplicates in the rule set.

hokageCV commented:
Updated the PR.  

In cssRules, colors were added but not being removed.  For example colors were generated thrice before copying, then 45 colors were copied. 15 for each color generation. 
Before generating new ones, have cleared them. Let me know your thoughts on it.


https://github.com/user-attachments/assets/e1163550-7d2c-42e4-8053-9bd2f49c5231

thisisfranciswu commented:
That's terrific, thanks for your contribution! I'm gonna go ahead and merge this.
